vpr/src/pack/cluster.cpp

```diff
@@ -300,6 +302,12 @@ void add_cluster_molecule_candidates_by_highfanout_connectivity(t_pb* cur_pb,
                                                                 const std::multimap<AtomBlockId, t_pack_molecule*>& atom_molecules,
                                                                 const int feasible_block_array_size);
 
+void add_cluster_molecule_candidates_by_attraction_group(t_pb* cur_pb,
```

Comment:
make the functions you are adding static (add to declaration and definition) as well - makes them file scope

Reply:
also add comments in the definitions (to match the other routines)

Reply:
add comments for ones you have figured out as well